### PR TITLE
[Vectorized][Feature] support money_format/ucase/character_length

### LINKED ISSUE #7648: [Vectorized][Feature] support money_format/ucase/character_length
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

support money_format/ucase/character_length

### Use case

```sql
select money_format(123);
select ucase('aa1');
select character_length('aa1');
```

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

Close related #7648 (replace it with issue number if it exists).

@HappenLee 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] I have created an issue on (Fix #7648) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -1038,4 +1041,129 @@ class FunctionStringParseUrl : public IFunction {
...
+    static Status execute_impl(FunctionContext* context, Block& block,
+                               const ColumnNumbers& arguments, size_t result,
+                               size_t input_rows_count) {
+        auto res_column = ColumnString::create();
+        ColumnPtr argument_column = block.get_by_position(arguments[0]).column;
+
+        auto data_column = assert_cast<const ColumnVector<Float64>*>(argument_column.get());
+        auto result_column = assert_cast<ColumnString*>(res_column.get());
+
+        for (int i = 0; i < input_rows_count; i++) {
```
Comment (zuochunwei) [inline be/src/vec/functions/function_string.h:1079]: make int i as size_t i

File: be/src/vec/functions/function_string.h [hunk 2]
```diff
@@ -1038,4 +1041,129 @@ class FunctionStringParseUrl : public IFunction {
...
+        block.replace_by_position(result, std::move(res_column));
+        return Status::OK();
+    }
+};
+
+struct MoneyFormatInt64Impl {
+    static DataTypes get_variadic_argument_types() { return {std::make_shared<DataTypeInt64>()}; }
+
+    static Status execute_impl(FunctionContext* context, Block& block,
+                               const ColumnNumbers& arguments, size_t result,
```
Comment (zuochunwei) [inline be/src/vec/functions/function_string.h:1095]: these execute_impl versions just have a little different.
suggest you to design a common function reducing code lines?

File: be/src/vec/functions/function_string.cpp [hunk 3]
```diff
@@ -272,7 +272,7 @@ struct HexStringName {
 
 struct HexStringImpl {
     static DataTypes get_variadic_argument_types() {
-        return {std::make_shared<vectorized::DataTypeString>()};
+        return {std::make_shared<DataTypeString>()};
```
Comment (spaces-X) [inline be/src/vec/functions/function_string.cpp:275]: test

File: be/src/vec/functions/function_string.h (hunk 2 above)
Comment (BiteTheDDDDt) [reply be/src/vec/functions/function_string.h]: > these execute_impl versions just have a little different. suggest you to design a common function reducing code lines?

I do some improve.

**Pull request merged**